table winners  and logic for automatically creating winner

janesvami opened the pull request:
Add a table "winners" and logic for automatically creating a winner at month-end closing. To do this, add methods for opening and closing weeks and months to MeasurementController and a complete package for the winner table: entity, repository, controller and service. Add to the database up-to-date information: user profiles, weeks and months

janesvami committed:
add a table "winners" and logic for automatically creating a winner at month-end closing. To do this, add methods for opening and closing weeks and months to MeasurementController and a complete package for the winner table: entity, repository, controller and service.

HailToSG commented on src/main/java/ani/beautymarathon/controller/MeasurementController.java:
```diff
@@ -140,6 +144,49 @@ public Page<GetUserMeasurementView> getAllMeasurements(
+    @Operation(summary = "Update month status by ID",
+            description = """
+                     This operation updates the status of the month for the given ID to either "OPEN" or "CLOSED".
+                    A "CLOSED" week is read-only.""",
```
a closer "WEEK"? Insn't it an endpoint for a MONTH?

janesvami replied:
Fixed

HailToSG commented on src/main/java/ani/beautymarathon/controller/WinnerController.java:
```diff
@@ -0,0 +1,23 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Tag(
+        name = "Winner controllers",
+        description = "Controllers for all operations with winners"
+)
+@RequestMapping("/winners")
+public class WinnerController {
+
+    private final WinnerService winnerService;
+
+    public WinnerController(WinnerService winnerService) {
+        this.winnerService = winnerService;
+    }
```
the controller for a nothing? Where is an at least one method?

janesvami replied:
I created it in advance, sorry.

HailToSG commented on src/main/java/ani/beautymarathon/service/MeasurementService.java:
```diff
@@ -125,6 +133,43 @@ public Page<MoMeasurement> getCascadeOfAllMeasurements(MoMeasurementFilter filte
+        winnerService.createWinnersFromViews(userMaxAverageViews, moMeasurement);
+        log.info("The winner of the month has been determined!");
+
+        return updatedMoMeasurement;
```
add here the same ClosedState value handling and all the discussed

janesvami replied:
Fixed

HailToSG commented on src/main/java/ani/beautymarathon/service/WinnerService.java:
```diff
@@ -0,0 +1,40 @@
+        List<Winner> winners = new ArrayList<>();
+
+        for (UserMaxAverageView userMax : userMaxAverageViews) {
+            Winner winner = new Winner();
```
got final?

janesvami replied:
Fixed

janesvami committed:
fixed

janesvami committed:
fix

janesvami committed:
add method updateUserMeasurement to MeasurementController
with view for updating, add column email to the table user_profile with unique index
